### PR TITLE
Added mythic plus condition and DogTag

### PR DESCRIPTION
**Pull request by arieh**
Since I've ported this from my addon it requires more testing, so this PR is more about getting comments.

### CODE REVIEW

File: Components/Core/Conditions/Categories/Location.lua [hunk 1]
```diff
@@ -172,8 +172,39 @@ ConditionCategory:RegisterCondition(1,	 "INSTANCE2", {
...
+    icon = "Interface\\Icons\\inv_relics_hourglass",
+    tcoords = CNDT.COMMON.standardtcoords,
+
+    Env = {
+    	GetActiveKeystoneInfo = function()
+    		return select(1, C_ChallengeMode.GetActiveKeystoneInfo()) or 0
+    	end
+	},
+
+    specificOperators = {["<="] = true, [">="] = true, ["=="]=true, ["~="]=true, [">"]=true, ["<"] = true},
```
Comment (ascott18) [inline Components/Core/Conditions/Categories/Location.lua]: Don't need to include `specificOperators` if all of them are enabled.

File: Components/Core/Conditions/Categories/Location.lua [hunk 2]
```diff
@@ -172,8 +172,39 @@ ConditionCategory:RegisterCondition(1,	 "INSTANCE2", {
...
+
+    applyDefaults = function(conditionData, conditionSettings)
```
Comment (ascott18) [inline Components/Core/Conditions/Categories/Location.lua]: Don't need `applyDefaults` either if not using `specificOperators`

File: Components/Core/Conditions/Categories/Location.lua [hunk 3]
```diff
@@ -172,8 +172,39 @@ ConditionCategory:RegisterCondition(1,	 "INSTANCE2", {
...
+    tooltip = L["CONDITIONPANEL_KEYSTONE_DESC"],
+    min = 0,
+    max = 30,
+    unit= false,
+    icon = "Interface\\Icons\\inv_relics_hourglass",
+    tcoords = CNDT.COMMON.standardtcoords,
+
+    Env = {
+    	GetActiveKeystoneInfo = function()
+    		return select(1, C_ChallengeMode.GetActiveKeystoneInfo()) or 0
```
Comment (ascott18) [inline Components/Core/Conditions/Categories/Location.lua]: Don't need to `select` the first return of a function - just return `C_ChallengeMode.GetActiveKeystoneInfo() or 0` directly. (I'm assuming that the signature is `local keystoneLevel, ... = C_ChallengeMode.GetActiveKeystoneInfo()`)

File: Components/Core/Conditions/Categories/Location.lua [hunk 4]
```diff
@@ -172,8 +172,39 @@ ConditionCategory:RegisterCondition(1,	 "INSTANCE2", {
 	end,
 })
 
+ConditionCategory:RegisterCondition(1.1,  "MYTHICKEYSTONE", {
```
Comment (ascott18) [inline Components/Core/Conditions/Categories/Location.lua]: I'd prefer `KEYSTONELEVEL` as the identifier. Its more descriptive of what's being checked - MYTHICKEYSTONE could refer to anything about a M+ keystone.

File: Localization/TellMeWhen-enUS.lua [hunk 5]
```diff
@@ -1915,6 +1917,8 @@ L["DT_DOC_ValueMax"] = "Returns the maximum of the numerical value that the icon
 L["DT_INSERTGUID_TOOLTIP"] = "|cff7fffffShift-click|r to insert this icon's identifier into a DogTag."
 L["DT_INSERTGUID_GENERIC_DESC"] = [[If you would like one icon to display information about another, |cff7fffffShift-click|r that icon to insert its unique identifier that you can pass as the tag's "icon" parameter.]]
 
+L["DT_KEYSTONE_DESC"] = "Level of the currently active mythic keystone"
```
Comment (ascott18) [inline Localization/TellMeWhen-enUS.lua]: Redundant with `CONDITIONPANEL_KEYSTONE_DESC` - don't need two of the same string. I have no qualms with using a string keyed as `CONDITIONPANEL` as a description for a DogTag. The localization keys are pretty much a free-for-all (unfortunately).

File: Components/Core/IconDataProcessors.lua [hunk 6]
```diff
@@ -752,6 +752,17 @@ do
 		example = '[MaxDuration] => "3"; [MaxDuration:TMWFormatDuration] => "3.0"; [MaxDuration(icon="TMW:icon:1I7MnrXDCz8T")] => "60"',
 		category = L["ICON"],
 	})
+
+	TMW.C.IconComponent:RegisterDogTag("TMW", "KeystoneLevel", {
```
Comment (ascott18) [inline Components/Core/IconDataProcessors.lua]: This doesn't really belong here. I'd say that this DogTag doesn't really belong in TMW at all, but rather in LDT itself.

If you open a separate issue on https://www.wowace.com/projects/libdogtag-3-0/issues and then link the issue you create to me, I can add this to LDT.

**Comment by arieh** [comment]
Updated PR with requested changes. I've also tested this to be working as intended

**Pull request merged**